src/Voron/Data/BTrees/Tree.cs

```diff
@@ -365,8 +365,14 @@ public DirectAddScope DirectAdd(Slice key, int len, out byte* ptr)
         {
             return DirectAdd(key, len, TreeNodeFlags.Data, out ptr);
         }
+        
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public DirectAddScope DataDirectAdd(Slice key, int len)
+        {
+            return DirectAdd(key, len, TreeNodeFlags.Data, out _, requiresDataPtr: false);
+        }
 
-        public DirectAddScope DirectAdd(Slice key, int len, TreeNodeFlags nodeType, out byte* ptr)
+        public DirectAddScope DirectAdd(Slice key, int len, TreeNodeFlags nodeType, out byte* ptr, bool requiresDataPtr = true)
```

Comment:
I have some issue with `requiresDataPtr` parameter name. I was thinking about the following options:

- `bool keyOnly = false`
- `bool populateDataPtr = true`

Reply:
I like `populateDataPtr` option. Added.